app/models/concerns/due_date_actions.rb

```diff
@@ -51,250 +51,45 @@ def drop_topic_permissible?
-      metareview: metareview_permissible?,
-      quiz: quiz_permissible?,
-      team_formation: team_formation_permissible?,
-      signup: signup_permissible?,
-      drop_topic: drop_topic_permissible?,
-      has_any_permissions: has_allowed_actions?
-    }
+    deadline.deadline_type&.name || 'unknown'
   end
 
-  # Topic-specific permission checking
-  def activity_permissible_for_topic?(activity, topic_id)
-    deadline = current_stage_for_topic(topic_id, activity)
-    return false unless deadline
-
-    deadline.activity_permissible?(activity)
-  end
-
-  # Topic-specific syntactic sugar methods
-  def submission_permissible_for_topic?(topic_id)
-    activity_permissible_for_topic?(:submission, topic_id)
-  end
-
-  def review_permissible_for_topic?(topic_id)
-    activity_permissible_for_topic?(:review, topic_id)
-  end
-
-  def quiz_permissible_for_topic?(topic_id)
-    activity_permissible_for_topic?(:quiz, topic_id)
-  end
-
-  # Copy all due dates to a new parent object
+  # Copy due dates to a new parent object
   def copy_due_dates_to(new_parent)
     due_dates.find_each do |due_date|
-      due_date.copy_to(new_parent)
-    end
-  end
-
-  # Duplicate due dates with modifications
-  def duplicate_due_dates_with_changes(new_parent, changes = {})
-    due_dates.map do |due_date|
-      due_date.duplicate_with_changes(changes.merge(parent: new_parent))
-    end
-  end
-
-  # Create a new due date for this parent
-  def create_due_date(deadline_type_name, due_at, round: 1, **attributes)
-    deadline_type = DeadlineType.find_by_name(deadline_type_name)
-    raise ArgumentError, "Invalid deadline type: #{deadline_type_name}" unless deadline_type
-
-    due_dates.create!(
-      deadline_type: deadline_type,
-      due_at: due_at,
-      round: round,
-      **attributes
-    )
-  end
-
-  # Update or create a due date for a specific type and round
-  def set_deadline(deadline_type_name, due_at, round: 1, **attributes)
-    deadline = find_deadline(deadline_type_name, round)
-
-    if deadline
-      deadline.update!(due_at: due_at, **attributes)
-      deadline
-    else
-      create_due_date(deadline_type_name, due_at, round: round, **attributes)
+      new_due_date = due_date.dup
+      new_due_date.parent = new_parent
+      new_due_date.save!
     end
   end
 
-  # Remove due dates of a specific type
-  def remove_deadlines_of_type(deadline_type_name)
-    due_dates.joins(:deadline_type)
-             .where(deadline_types: { name: deadline_type_name })
-             .destroy_all
-  end
-
-  # Shift all deadlines by a certain amount of time
-  def shift_deadlines(time_delta)
-    due_dates.update_all("due_at = due_at + INTERVAL #{time_delta.to_i} SECOND")
-  end
-
-  # Shift deadlines of a specific type
-  def shift_deadlines_of_type(deadline_type_name, time_delta)
+  # Shift deadlines of a specific type by a time interval
+  def shift_deadlines_of_type(deadline_type_name, time_interval)
```

Comment:
I would only expect the time interval to be expressed in DAYS.  I think that using "days" as identifier names and in the calculation would improve the readability of this method.